### PR TITLE
Add generic grid builder and refactor builder code that uses generic grids

### PR DESCRIPTION
**Pull request by sethrj**
This adds a helper class for building `GenericGridData` (which we can later rename to `GenericGridRecord` and probably consolidate some of the grid classes). It removes the unused "generic grid" value grid builder/inserter functions for now, and also removes the unsupported "interp" data for the grids, since those are also unused.

I also added some error handling code for the livermore PE reader and inserter, since hydrogen doesn't have low-energy cross sections.

### CODE REVIEW

File: src/celeritas/io/LivermorePEReader.cc [hunk 1]
```diff
@@ -95,25 +97,32 @@ LivermorePEReader::operator()(AtomicNumber atomic_number) const
...
+            for (int i = 0; i < size; ++i)
             {
                 CELER_ASSERT(infile);
                 infile >> result.xs_lo.x[i] >> result.xs_lo.y[i];
             }
         }
+        else if (atomic_number != AtomicNumber{1})
+        {
+            // Hydrogen doesn't have low-energy cross sections
+            CELER_LOG(warning) << "No low-energy cross sections found in '"
```
Comment (amandalund) [inline src/celeritas/io/LivermorePEReader.cc]: Total cross sections below the K-shell energy aren't present for elements with only one subshell, so this warning should be for Z > 2.

File: test/celeritas/grid/ValueGridBuilder.test.cc [hunk 2]
```diff
@@ -123,30 +123,6 @@ TEST_F(ValueGridBuilderTest, log_grid)
     }
 }
 
-TEST_F(ValueGridBuilderTest, DISABLED_generic_grid)
```
Comment (amandalund) [inline test/celeritas/grid/ValueGridBuilder.test.cc:126]: Maybe we should add a test for the new `GenericGridBuilder`?

**Review by whokion — APPROVED** [review]
Thanks for the work.  Probably @amandalund has more feedback.

File: src/celeritas/em/model/LivermorePEModel.cc [hunk 3]
```diff
@@ -21,16 +21,18 @@
 #include "celeritas/global/CoreParams.hh"
 #include "celeritas/global/CoreState.hh"
 #include "celeritas/global/TrackExecutor.hh"
+#include "celeritas/grid/GenericGridBuilder.hh"
```
Comment (amandalund) [inline src/celeritas/em/model/LivermorePEModel.cc]:
```suggestion
```

File: src/celeritas/em/model/detail/LivermoreXsInserter.hh [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        }
+    }
+    using units::MevEnergy;
+
+    LivermoreElement el;
+
+    // Add tabulated total cross sections
+    if (inp.xs_lo)
+    {
+        // Hydrogen doesn't have low-energy cross sections
```
Comment (amandalund) [inline src/celeritas/em/model/detail/LivermoreXsInserter.hh]:
```suggestion
        // Z < 3 don't have low-energy cross sections
```

File: src/celeritas/phys/detail/ParticleInserter.hh [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Add a particle.
+ */
+auto ParticleInserter::operator()(Input const& inp) -> Id
+{
+    CELER_EXPECT(inp.mass >= zero_quantity());
+    CELER_EXPECT(inp.decay_constant >= 0);
```
Comment (amandalund) [inline src/celeritas/phys/detail/ParticleInserter.hh]: Should we be `CELER_VALIDATE`-ing these?

**Review by amandalund — APPROVED** [review]
Looks good, thanks @sethrj!

File: src/celeritas/phys/detail/ParticleInserter.hh (hunk 5 above)
Comment (sethrj) [reply src/celeritas/phys/detail/ParticleInserter.hh]: Good point, I just moved these from `ParticleParams::ParticleParams` but yes we should.

**Pull request merged**